### PR TITLE
TECH-558: Added cucumber tests for GET /services

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
**Description:**
Add cucumber test for GET /services.

Ticket: https://govstack-global.atlassian.net/browse/TECH-558

### CODE REVIEW

**Comment by KarolinaKopacz** [comment]
@ingmarvali 
The `serviceKey` property is required in the schema, but there is no serviceKey in the properties - does that mean that `id` should be required because the id description contains information about the service key?
If so, I can update the API spec and change `serviceKey` to `id` in the list of required fields

Zrzut ekranu 2023-04-17 o 15 34 11

File: test/openAPI/features/services_services.feature [hunk 1]
```diff
@@ -0,0 +1,30 @@
+@method=GET @endpoint=/services
+Feature: API endpoint that returns a list of services (Services available to use)
+  
+  Listing of all services with basic information.
+
+  @smoke
+  Scenario: Retrieve the list of services smoke type test
+
+    Given User wants to retrieve the list of services
+    When User sends GET request with given "Register Mother and Child" as name
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/services_services.feature]: smoke test should be invoked without optional parameters to retrieve a list of all services.

File: test/openAPI/features/services_services.feature [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    Then The response from the /statistics endpoint is received
+    And The /statistics response should be returned in a timely manner 15000ms
+    And The /statistics response should have status 200
+    And The /statistics response should match json schema
+
+  @unit @positive
+  Scenario Outline: Retrieve the list of services
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/services_services.feature]: Here i believe only those services are returned that match the name provided in optional parameter. I would change the scenario name to be more descriptive

File: test/openAPI/features/services_services.feature [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    Given User wants to retrieve the list of services
+    When User sends GET request with given "<name>" as name
+    Then The response from the /statistics endpoint is received
+    And The /statistics response should be returned in a timely manner 15000ms
+    And The /statistics response should have status 200
+    And The /statistics response should match json schema
+
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/services_services.feature:26]: We should also check here if services that we received (if any) match with the name that we provided in the optional parameter

File: test/openAPI/features/support/services_services.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+const chai = require('chai');
+const { spec } = require('pactum');
+const { Given, When, Then, Before } = require('@cucumber/cucumber');
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/services_services.js]: Missing After with its code

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]

File: test/openAPI/features/services_services.feature [hunk 5]
```diff
@@ -0,0 +1,31 @@
+@method=GET @endpoint=/services
+Feature: API endpoint that returns a list of services (Services available to use)
+  
+  Listing of all services with basic information.
+
+  @smoke
+  Scenario: Retrieve the list of services smoke type test
+
+    Given User wants to retrieve the list of services
+    When User sends GET /statistics request with no optional parameter
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/services_services.feature]:
```suggestion
    When GET request to retrieve a list of services is sent
```
I think it sounds better than with no optional parameter

File: test/openAPI/features/services_services.feature [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  @unit @positive
+  Scenario Outline: Retrieve the list of services that match the name provided in optional parameter
+
+    Given User wants to retrieve the list of services
+    When User sends GET /statistics request with given "<name>" as name
+    Then The response from the /statistics endpoint is received
+    And The /statistics response should be returned in a timely manner 15000ms
+    And The /statistics response should have status 200
+    And The /statistics response should match json schema
+    And The service name in response match provided "<name>" as name parameter
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/services_services.feature]:
```suggestion
       And The name field in the response matches with "<name>" provided as name in the query parameter
       or simply
       And The response is filtered by "<name>" provided in the optional parameter
```

or something similar

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]
Except those 2 comments. Have you checked if 
```
 And The service name in response match provided "<name>" as name parameter
```
will check all the elements returned. Or is it going to check just the first one. I believe there can be more objects returned with the same name but different versions. Please confirm that it will work for both cases (one returned element and more than one)

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]
I like the approach. I saw just 2 more issues.

1. It will always fail when there will be no element returned, should we skip the test? or we assume that there will be always at least 1 element matching the name in the request. What do you think about it @KarolinaKopacz ?

2. You used .includes and I assume that is why:  
image

When you will put When - which is invoking .withQueryParams into brackets {} it should return single value: 
image

Don't ask me why, but it seems it works like that.

**Review by mgrochalskisoldevelo — APPROVED** [review]

**Review by ingmarvali — APPROVED** [review]
seems ok

**Pull request merged**